test/extended/ci/job_names.go

```diff
@@ -33,6 +33,17 @@ var _ = g.Describe("[sig-ci] [Early] prow job name", func() {
 	}
 	isPeriodic := strings.HasPrefix(jobName, "periodic-")
 
+	g.It("should match feature set", func() {
+		if jobName == "" {
+			e2eskipper.Skipf("JOB_NAME env var not set, skipping")
+		}
+
+		isTechPreview := exutil.IsTechPreviewNoUpgrade(oc)
+		if strings.Contains(jobName, "-techpreview") && !isTechPreview {
```

Comment:
Check in the other direction too. If the job is tech review, it should be in the name. Right?